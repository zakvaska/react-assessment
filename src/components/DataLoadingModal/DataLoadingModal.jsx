import React from 'react';
import PropTypes from 'prop-types';
import Modal from '../Modal/Modal';
import { Spinner, SpinnerWrapper, CenteredLabel } from '../styled-components/Task';

const DataLoadingModal = ({ modalCloseHandler }) => (
  <Modal modalCloseHandler={modalCloseHandler} color="#4b8ed2">
    <SpinnerWrapper>
      <Spinner data-testid="modal" />
    </SpinnerWrapper>
    <CenteredLabel>Data is loading</CenteredLabel>
  </Modal>
);

DataLoadingModal.propTypes = {
  modalCloseHandler: PropTypes.func.isRequired
};

export default DataLoadingModal;
