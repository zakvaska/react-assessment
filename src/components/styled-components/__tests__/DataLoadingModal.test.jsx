import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import DataLoadingModal from '../../DataLoadingModal/DataLoadingModal';
import 'jest-styled-components';

it('DataLoadingModal renders correctly', async () => {
  const handleClose = jest.fn();
  const { container } = render(<DataLoadingModal modalCloseHandler={handleClose} />);
  expect(container).toMatchSnapshot();
  fireEvent.keyDown(window, { key: 'Escape' });
  expect(handleClose).toHaveBeenCalledTimes(1);
});
