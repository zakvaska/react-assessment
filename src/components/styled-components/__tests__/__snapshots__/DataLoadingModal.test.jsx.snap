// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataLoadingModal renders correctly 1`] = `
.c0 {
  position: absolute;
  width: 100%;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
}

.c1 {
  background-color: white;
  border: 1px solid #d10720;
  width: 35rem;
  border-radius: .75rem;
  margin: 10rem auto;
  padding: 1.5rem;
  color: #4b8ed2;
  text-align: start;
  font-family: sans-serif;
  font-size: 1.1rem;
  cursor: default;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  display: inline-block;
  width: 3rem;
  height: 3rem;
  margin: 1rem auto;
  vertical-align: text-bottom;
  border: 0.5em solid #23619a;
  border-right-color: transparent;
  border-radius: 50%;
  -webkit-animation: cnyVFc 1.5s linear infinite;
  animation: cnyVFc 1.5s linear infinite;
}

.c4 {
  text-align: center;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
          data-testid="modal"
        />
      </div>
      <div
        class="c4"
      >
        Data is loading
      </div>
    </div>
  </div>
</div>
`;
